alibi/explainers/ale.py

```diff
@@ -143,10 +150,15 @@ def explain(self, X: np.ndarray, features: Optional[List[int]] = None, min_bin_p
 
         # TODO: use joblib to paralelise?
         for feature in features:
+            if grid_points is not None:
+                feature_grid_points = grid_points[feature_names[feature]]
```

Comment:
We need to do some validation here or wrap in `try` `except` as the user might not provide a fully specified dictionary.

Also, according to the discussion on the issue, it should be desirable to allow partially specified dictionaries and infer the points for other features.

Reply:
Instead of `try - except`, maybe we can do something like:
* initialize at the very beginning of the function
```python
if grid_points is None:
   grid_points = {}
```
* obtain the value through `get`
```python
feature_grid_points = grid_points.get(feature_names[feature])   # returns None by default if key not in dict
```

Reply:
Ok, I get the point. yes, using get should work